manifests/producer/service.pp

```diff
@@ -23,15 +23,14 @@
   $producer_jmx_opts                         = $kafka::producer::producer_jmx_opts,
   $producer_log4j_opts                       = $kafka::producer::producer_log4j_opts,
   $service_config                            = $kafka::producer::service_config,
-  $service_defaults                          = $kafka::producer::service_defaults,
 ) {
 
   if $caller_module_name != $module_name {
     fail("Use of private class ${name} by ${caller_module_name}")
   }
 
   if $service_install {
-    $producer_service_config = deep_merge($service_defaults, $service_config)
+    $producer_service_config = $service_config
```

Comment:
is this assignment still needed? Can't we just work with $service_config?

Reply:
It is currently needed because `$producer_service_config` is currently used in the ERB templates.

Cleaning this is next on my todo list ;-)